common/app/common/Logging.scala

```diff
@@ -25,6 +29,26 @@ trait Logging {
   def logErrorWithCustomFields(message: String, error: Throwable, customFields: List[LogField]): Unit = {
     log.logger.error(customFieldMarkers(customFields), message, error)
   }
+
+  def errorLoggingF[A](message: String)(task: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    def errorLoggingFuture(f: Future[A]): Future[A] = {
+      f.recover {
+        case NonFatal(e) => Logger.error(message, e) ; throw e
```

Comment:
generally I try to avoid throwing exceptions in scala, as it is uncertain who would deal with the fallout (not really FP)
However I appreciate that might involve changing types in various places, and you can never stop other code throwing either.

Reply:
It's not creating a new exception. 

This is just a helper for nicer syntax around error logging. It tries to be transparent, passing the error on after logging it with some context. For example, it makes it easy to log all the function parameters on an exception.

A lot of places can throw exceptions in the Guardian frontend project since it uses Play framework and the Java Amazon API directly.